examples/DynamicCantileverCase/dynamic_cantilever_helper.py

```diff
@@ -0,0 +1,112 @@
+        self.beta = betas[mode]
+        self.omega = (self.beta ** 2) * np.sqrt(
+            youngs_modulus
+            * moment_of_inertia
+            / (density * base_area * base_length ** 4)
+        )
+
+        np_mode_l = self._compute_nonparametrized_mode(
+            base_length, base_length, self.beta
+        )
+        self.param = end_velocity_z / (-1j * self.omega * np_mode_l)
+
+        # if mode > 1:
+        #     step = 1.0 / (mode - 1)
+        #     test_x = np.arange(0.0, 1.0 + step, step=step)[1:] * base_length
+        #     param_mat = np.zeros([mode, mode], dtype="complex128")
+        #     target_mat = np.zeros([mode, 1], dtype="complex128")
+        #
+        #     for i, x in np.ndenumerate(test_x):
+        #         param_mat[i, :] = self._compute_nonparametrized_mode(x, base_length, self.betas)
+        #         target_mat[i] = 0.0
+        #
+        #     param_mat[-1, :] = -1j * self.omegas * np_modes_l
+        #     target_mat[-1] = end_velocity_z
+        #
+        #     soln_mat = la.solve(param_mat, target_mat)
+        #     self.params = soln_mat.ravel()
+
+    def get_initial_position_profile(self, x_coords):
+        positions = []
+
+        for x in x_coords:
+            positions.append(
+                np.real(
+                    self.param
+                    * self._compute_nonparametrized_mode(
+                        self.base_length * x, self.base_length, self.beta
+                    )
+                )
+            )
+        return np.array(positions)
+
+    def get_initial_velocity_profile(self, x_coords):
+        velocities = []
+
+        for x in x_coords:
+            velocities.append(
+                np.real(
+                    (-1j * self.omega * self.param)
+                    * self._compute_nonparametrized_mode(
+                        self.base_length * x, self.base_length, self.beta
+                    )
+                )
+            )
+        return np.array(velocities)
+
+    def get_positions(self, x_coord, time_array):
+        positions = (
+            self.param
+            * self._compute_nonparametrized_mode(
+                self.base_length * x_coord, self.base_length, self.beta
+            )
+            * np.exp(-1j * self.omega * time_array)
+        )
+        return np.real(positions)
+
+    def get_velocities(self, x_coord, time_array):
+        velocities = (
+            (-1j * self.omega * self.param)
+            * self._compute_nonparametrized_mode(
+                self.base_length * x_coord, self.base_length, self.beta
+            )
+            * np.exp(-1j * self.omega * time_array)
+        )
+        return np.real(velocities)
+
+    def get_omega(self):
+        return self.omega
+
+    def get_amplitude(self):
+        return self.end_velocity_z / self.omega
+
+    @staticmethod
+    def _compute_nonparametrized_mode(x, base_length, betas):
```

Comment:
Why is this `betas`? I thought it is scalar. Since depending on the mode you want, you have a single beta value. So we can change to `beta`

Reply:
[3e1ca62](https://github.com/GazzolaLab/PyElastica/pull/173/commits/3e1ca62f44482de6dc719256fd47975ea3aa5037)